server/update-user.js

```diff
@@ -49,7 +49,7 @@ export default function updateUserFactory(analyticsClient) {
       anonymousId = _.first(user.anonymous_ids);
     }
 
-    const publicIdField = public_id_field === "id" ? "id" : "external_id";
+    const publicIdField = public_id_field; // === "id" ? "id" : "external_id";
```

Comment:
Yes we can remove the commented part. So the `ignore_segment_userId` is only for incoming data while the `public_id_field` is for outgoing. You can use both independently, but I agree, we should safe-guard this, just in case the `public_id_field` is not set, that way we ensure backwards compatibility. Good catch!